functions-and-scope.js

```diff
@@ -0,0 +1,199 @@
+// Log het antwoord in de terminal.
+
+let averageMark = 0
+for (let i = 0; i < grades.length; i++) {
+    if ([i] < grades.length -1 ) {
+        averageMark = averageMark + grades[i]
+    } else {
+        averageMark = averageMark + grades[i]
+        averageMark =  averageMark / grades.length
+    }
+
+}
+
+console.log(averageMark)
+
+
+
+
+// ---- Verwachte uitkomst: 6.642857142857143
+
+
+/* 2b: Omschrijven tot een herbruikbare functie */
+// Schrijf een functie genaamd averageGrade, die een array van cijfers verwacht (zoals grades) en het gemiddelde cijfer teruggeeft. Gebruik hiervoor jouw antwoord van 2a.
+// Zorg ervoor dat jouw functie ook werkt als we een andere array willen checken, zoals bijvoorbeeld: [6, 4, 5] of [8, 9, 4, 6, 10].
+// Log het antwoord in de terminal.
+
+let averageMark2 = 0;
+function averageGrade (averageCalculate){
+    for (let i = 0; i < averageCalculate.length; i++) {
+        if ([i] < averageCalculate.length -1 ) {
+            averageMark2 = averageMark2 + averageCalculate[i]
+        } else {
+            averageMark2 = averageMark2 + averageCalculate[i]
+            averageMark2 =  averageMark2 / averageCalculate.length
+        }
+    }
+    return averageMark2
+}
+
+const outcome3 = averageGrade(grades)
+console.log(outcome3)
+
+// ---- Verwachte uitkomsten:
+// averageGrade(grades) geeft 6.642857142857143
+// averageGrade([6, 4, 5]) geeft xxxx
+// averageGrade([8, 9, 4, 6, 10]) geeft xxxx
+
+
+/* 2c: Afronden op twee decimalen */
+// Zorg ervoor dat het gemiddelde cijfer dat wordt teruggegeven uit de functie netjes wordt afgerond op twee decimalen.
+// Tip: Google is your best friend!
+
+const decimals = outcome3.toFixed(2)
+console.log(decimals)
+
+/* Bonusopdracht: hoogste cijfer */
+
+/* 3a: Script schrijven  */
+// Schrijf een script die op basis van de grades array (hierboven) checkt wat het hoogst behaalde cijfer is. Je mag hier geen bestaande methoden voor gebruiken. Schrijf de stapjes eerst uit en vraag jezelf de volgende dingen af:
+// * Hoe kan ik iedere waarde van de array langsgaan?
+// * Op welke conditie moet ik checken?
+// * Hoe zorgt ik ervoor dat wanneer ik een cijfer tegenkom die aan de conditie voldoet, ik dit ergens kan opslaan?
+// Log het antwoord in de terminal.
+
+let highestMark = 0
+
+for (let i = 0; i < grades.length; i++) {
+    if (grades[i] > 9 ) {
+        highestMark = 10
+    } else if (grades.length > 8){
+        highestMark = 9
+    } else if (grades.length > 7){
+        highestMark = 8
+    } else if (grades.length > 6 ){
+        highestMark = 7
+    } else if (grades.length > 5){
+        highestMark = 6
+    } else if (grades.length > 4){
+        highestMark = 5
+    } else if (grades.length > 3){
+        highestMark = 4
+    } else if (grades.length > 2){
+        highestMark = 3
+    } else if (grades.length > 1){
+        highestMark = 2
+    }else {
+        highestMark = 1
+    }
```

Comment:
Leuk bedacht, maar wel een flink lap code! Nu willen we alleen het uiterste hoogste cijfer van een array terug krijgen. De code kan daarom dus een stuk korter zijn:

`let highestMark = 0

for(let i = 0; i < grades.length; i++) {
if(grades[i] > highestMark) {
      highestMark = grades[i];
}
}

console.log(highestMark);
`